### PR TITLE
FT: enable usEastBehavior for abortmpu result

### LINKED ISSUE #568: Abort MPU returns 404 for all regions when Upload Id doesn't exist
Tests reveal that AWS S3 returns 404 for us-east-1 and 204 for all other regions. We have a customer complaining about this incorrect behavior. We could leverage `usEastBehavior` to toggle the legacy behavior.
The fix should be against `6.2.5`.

### PR DESCRIPTION
**Pull request by electrachong**
FIXES #568

### CODE REVIEW

**Comment by electrachong** [comment]
end-to-end: http://ci.ironmann.io/gh/scality/Integration/9343

**Comment by electrachong** [comment]
new end-to-end: http://ci.ironmann.io/gh/scality/Integration/9345

**Comment by electrachong** [comment]
end-to-end failed due to completempu issues (https://github.com/scality/Integration/issues/449);
retrying: http://ci.ironmann.io/gh/scality/Integration/9409

File: lib/api/multipartDelete.js [hunk 1]
```diff
@@ -54,17 +55,25 @@ function multipartDelete(authInfo, request, log, callback) {
...
-                        // usEastBehavior is enabled in config
+                        // if legacy behavior is enabled for 'us-east-1' and
+                        // request is from 'us-east-1', return 404 instead of
+                        // 204
+                        if (config.usEastBehavior &&
+                            locationConstraint === 'us-east-1') {
+                            return callback(err);
+                        }
+                        // otherwise ignore error and return 204 status code
                         return callback();
```
Comment (rahulreddy) [inline lib/api/multipartDelete.js]: It's interesting that we don't return error in the callback if it's any error other than `errors.NoSuchUpload`. We have to fix this, may be in a different PR.

File: tests/functional/aws-node-sdk/test/object/deleteMpu.js [hunk 2]
```diff
@@ -46,7 +23,75 @@ describe('DELETE multipart', () => {
                     `Found unexpected statusCode ${statusCode}`);
                 assert.strictEqual(err, null,
                     `Expected no err but found ${err}`);
-                done(err);
+                callback(err);
+            });
+        }
+
+        it('on bucket that does not exist: should return NoSuchBucket',
```
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/object/deleteMpu.js]: @rahulreddy : confirmed aws returns NoSuchBucket for region other than us-east-1:
```
$ aws s3api  abort-multipart-upload --bucket nonexistingbucket --key objectKey --upload-id nonexistingid --region us-west-1

An error occurred (NoSuchBucket) when calling the AbortMultipartUpload operation: The specified bucket does not exist
```

**Comment by electrachong** [comment]
end-to-end passed

File: tests/functional/aws-node-sdk/test/object/deleteMpu.js (hunk 2 above)
Comment (rahulreddy) [inline tests/functional/aws-node-sdk/test/object/deleteMpu.js]: @electrachong Thanks!

**Pull request merged**